e2e/optimism_test.go

```diff
@@ -204,6 +204,17 @@ func testOptimismGenericCommitment(t *testing.T, v2Enabled bool) {
 	ot := actions.NewDefaultTesting(t)
 
 	optimism := NewL2AltDA(ot, proxyTS.Address(), true)
+	exerciseGenericCommitments(t, ot, proxyTS, optimism)
+}
+
+// TODO: @ethen can you help me formulate a description of what this sequence of calls is doing?
+func exerciseGenericCommitments(
+	t *testing.T,
+	ot actions.StatefulTesting,
+	proxyTS testutils.TestSuite,
+	optimism *L2AltDA,
+) {
```

Comment:
let's try to refactor this function into smaller understandable parts. For example feels like we should be able to extract a "buildL1Block" function, but right now block1 and block2 are being built differently. Probably can refactor and extract out the FinalizationSignal in block2.

Also I personally think the require() should be moved out of this function and into tests. I prefer to see all the require statements when looking at a test. Helps see with a single look what exactly is being tested.

Reply:
[9bfbda9](https://github.com/Layr-Labs/eigenda-proxy/pull/342/commits/9bfbda9538fd0c3d2563f7f2c0fc3fef02f30ed2)

Made a few tweaks, and left a note to improve further in the future